src/lib.rs

```diff
@@ -254,6 +254,7 @@ pub struct Relation {
     pub kind: RelationKind,
     pub from: Id,
     pub to: Id,
+    pub id: u32,
```

Comment:
This looks like it should be in `RelationKind::Impl` rather than `Relation`, since `SuperTrait` won't have such an id